cstar/roms/simulation.py

```diff
@@ -1408,14 +1389,19 @@ def run(
                 + "\n If you have already run Simulation.build(), either run it again or "
                 + " add the executable path manually using Simulation.exe_path='YOUR/PATH'."
             )
+
         if self.discretization.n_procs_tot is None:
             raise ValueError(
                 "Unable to calculate node distribution for this Simulation. "
                 + "Simulation.n_procs_tot is not set"
             )
 
-        run_length_seconds = int((self.end_date - self.start_date).total_seconds())
-        n_time_steps = run_length_seconds // self.discretization.time_step
+        if self.runtime_code.working_path is None:
+            raise FileNotFoundError(
+                "Local copy of ROMSSimulation.runtime_code does not exist. "
+                + "Call ROMSSimulation.setup() or ROMSSimulation.runtime_code.get() "
```

Comment:
unnecessary string concatenations should be replaced with implicit concatenations (no need for +)